tutorials/stats/1_extractor_functions.py

```diff
@@ -0,0 +1,109 @@
+# %% [markdown]
+"""
+# 1. Extractor Functions
+
+The following example covers the basics of using the `stats` module.
+
+Statistics are collected from pipeline services by extractor functions
+that report the state of one or more pipeline components. The `stats` module
+provides several default extractors, but users are free to define their own
+extractor functions.
+
+It is required that the extractors have the following uniform signature:
+the expected arguments are always `Context`, `Pipeline`, and `ExtraHandlerRuntimeInfo`,
+while the expected return value is an arbitrary `dict` or a `None`. It is a preferred practice
+to define extractors as asynchronous functions.
+
+The output of these functions will be captured by an OpenTelemetry instrumentor and directed to
+the Opentelemetry collector server which in its turn batches and persists data
+to Clickhouse or other OLAP storages.
+For more information on OpenTelemetry instrumentation,
+refer to the body of this tutorial as well as OpenTelemetry documentation.
+
+"""
```

Comment:
Maybe we should add some `apiref` links to documentation?

Reply:
Done